nflow-engine/src/main/java/io/nflow/engine/internal/dao/WorkflowInstanceDao.java

```diff
@@ -484,13 +484,21 @@ private boolean addExpectedStatesToQueryAndUpdate(StringBuilder sql, long workfl
     return jdbc.update(sql.toString(), args) == 1;
   }
 
-  public WorkflowInstance getWorkflowInstance(int id) {
+  public WorkflowInstance getWorkflowInstance(int id, boolean includeChildWorkflowIds, boolean includeCurrentStateVariables,
```

Comment:
Alternative form for this method signature, which would not break if we invent more include-types.
```
  public enum WorkflowInstanceEnrichment {
    CHILD_WORKFLOW_IDS,
    CURRENT_STATE_VARIABLES,
    ACTIONS,
    STATE_VARIABLES
  }
  ...
  public WorkflowInstance getWorkflowInstance(int id, List<WorkflowInstanceEnrichment> enrichments, Long maxActions)
  ...
  WorkflowInstance instance = workflowInstances.getWorkflowInstance(instanceId, asList(CHILD_WORKFLOW_IDS, CURRENT_STATE_VARIABLES), null);
```